plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/version/KubernetesVersionManagerImpl.java

```diff
@@ -163,47 +169,15 @@ private VirtualMachineTemplate registerKubernetesVersionIso(final String version
             f.setAccessible(true);
             f.set(registerIsoCmd, isoChecksum);
         }
+        f = registerIsoCmd.getClass().getDeclaredField("accountName");
+        f.setAccessible(true);
+        f.set(registerIsoCmd, accountManager.getSystemAccount().getAccountName());
+        f = registerIsoCmd.getClass().getDeclaredField("domainId");
+        f.setAccessible(true);
+        f.set(registerIsoCmd, accountManager.getSystemAccount().getDomainId());
         return templateService.registerIso(registerIsoCmd);
     }
```

Comment:
I must have mist this before, but i would add setters or add an appropriate constructor instead of this tinkering with access.